client/apps/edit/components/admin/index.coffee

```diff
@@ -152,3 +157,42 @@ module.exports = class EditAdmin extends Backbone.View
     id = $(e.currentTarget).attr 'data-id'
     @article['featured' + resource].remove id
     @article.save()
+
+  setupSlug: ->
+    if $('.edit-admin-slug-input').val() is @generateSlugFromTitle()
+      $('.edit-admin-slug-generate').hide()
+    else
+      $('.edit-admin-slug-generate').show()
+
+    $('.edit-admin-slug-input').on 'keyup', _.debounce( =>
+      @slugify()
+      @article.save slug: $('.edit-admin-slug-input').val()
+    , 1000 )
+
+  slugify: ->
+    t = $('.edit-admin-slug-input')
+    slug = _s.slugify(t.val())
+    t.val slug
+
+  generateSlugFromTitle: ->
+    cat = [@article.get('author').name, @article.get('title')].join('-')
```

Comment:
Hmm seeing this duplicated logic server/client makes me wonder if the API should just let the client decide the slug and throw a validation error if a client tries to save an article without one... :smile_cat: :thought_balloon:  (out of scope for this PR)